compiler/runtime/OMRRelocationRecord.cpp

```diff
@@ -0,0 +1,393 @@
+   {
+   reloTarget->storeUnsigned16b(size,(uint8_t *) &_record->_size);
+   }
+
+uint16_t
+OMR::RelocationRecord::size(TR::RelocationTarget *reloTarget)
+   {
+   return reloTarget->loadUnsigned16b((uint8_t *) &_record->_size);
+   }
+
+
+void
+OMR::RelocationRecord::setType(TR::RelocationTarget *reloTarget, OMR::RelocationRecordType type)
+   {
+   reloTarget->storeUnsigned8b(type, (uint8_t *) &_record->_type);
+   }
+
+OMR::RelocationRecordType
+OMR::RelocationRecord::type(TR::RelocationTarget *reloTarget)
+   {
+   return (OMR::RelocationRecordType)_record->type(reloTarget);
+   }
+
+
+void
+OMR::RelocationRecord::setWideOffsets(TR::RelocationTarget *reloTarget)
+   {
+   self()->setFlag(reloTarget, FLAGS_RELOCATION_WIDE_OFFSETS);
+   }
+
+bool
+OMR::RelocationRecord::wideOffsets(TR::RelocationTarget *reloTarget)
+   {
+   return (self()->flags(reloTarget) & FLAGS_RELOCATION_WIDE_OFFSETS) != 0;
+   }
+
+void
+OMR::RelocationRecord::setPcRelative(TR::RelocationTarget *reloTarget)
+   {
+   self()->setFlag(reloTarget, FLAGS_RELOCATION_EIP_OFFSET);
+   }
+
+bool
+OMR::RelocationRecord::pcRelative(TR::RelocationTarget *reloTarget)
+   {
+   return (self()->flags(reloTarget) & FLAGS_RELOCATION_EIP_OFFSET) != 0;
+   }
+
+void
+OMR::RelocationRecord::setFlag(TR::RelocationTarget *reloTarget, uint8_t flag)
+   {
+   uint8_t flags = reloTarget->loadUnsigned8b((uint8_t *) &_record->_flags) | (flag & FLAGS_RELOCATION_FLAG_MASK);
+   reloTarget->storeUnsigned8b(flags, (uint8_t *) &_record->_flags);
+   }
+
+uint8_t
+OMR::RelocationRecord::flags(TR::RelocationTarget *reloTarget)
+   {
+   return reloTarget->loadUnsigned8b((uint8_t *) &_record->_flags) & FLAGS_RELOCATION_FLAG_MASK;
+   }
+
+void
+OMR::RelocationRecord::setReloFlags(TR::RelocationTarget *reloTarget, uint8_t reloFlags)
+   {
+   TR_ASSERT((reloFlags & ~FLAGS_RELOCATION_FLAG_MASK) == 0,  "reloFlags bits overlap cross-platform flags bits\n");
+   uint8_t crossPlatFlags = self()->flags(reloTarget);
+   uint8_t flags = crossPlatFlags | (reloFlags & ~FLAGS_RELOCATION_FLAG_MASK);
+   reloTarget->storeUnsigned8b(flags, (uint8_t *) &_record->_flags);
+   }
+
+uint8_t
+OMR::RelocationRecord::reloFlags(TR::RelocationTarget *reloTarget)
+   {
+   return reloTarget->loadUnsigned8b((uint8_t *) &_record->_flags) & ~FLAGS_RELOCATION_FLAG_MASK;
+   }
+
+void
+OMR::RelocationRecord::preparePrivateData(TR::RelocationRuntime *reloRuntime, TR::RelocationTarget *reloTarget)
+   {
+
+   }
+
+#undef FLAGS_RELOCATION_WIDE_OFFSETS
+#undef FLAGS_RELOCATION_EIP_OFFSET
+#undef FLAGS_RELOCATION_ORDERED_PAIR
+#undef FLAGS_RELOCATION_TYPE_MASK
+#undef FLAGS_RELOCATION_FLAG_MASK
+
+
+bool
+OMR::RelocationRecord::ignore(TR::RelocationRuntime *reloRuntime)
+   {
+   return false;
+   }
+
+int32_t
+OMR::RelocationRecord::applyRelocationAtAllOffsets(TR::RelocationRuntime *reloRuntime, TR::RelocationTarget *reloTarget, uint8_t *reloOrigin)
+   {
+   if (self()->ignore(reloRuntime))
+      {
+      return 0;
+      }
```

Comment:
You don't need to check again if it's already being checked in `handleRelocation`.

Reply:
Thank you for suggestion, I will remove these lines in the next commit